File: dl.py
import urllib.request
import os

try:
    os.makedirs('public/topojson', exist_ok=True)
    url = 'https://raw.githubusercontent.com/deldersveld/topojson/master/countries/thailand/thailand-provinces.json'
    urllib.request.urlretrieve(url, 'public/topojson/thailand-provinces.topojson')
    with open('dl_status.txt', 'w') as f:
        f.write('Success')
except Exception as e:
    with open('dl_status.txt', 'w') as f:
        f.write(str(e))

File: dl.js
const fs = require('fs');
const https = require('https');

if (!fs.existsSync('public')) {
    fs.mkdirSync('public');
}
if (!fs.existsSync('public/topojson')) {
    fs.mkdirSync('public/topojson');
}

const url = 'https://raw.githubusercontent.com/deldersveld/topojson/master/countries/thailand/thailand-provinces.json';

const file = fs.createWriteStream('public/topojson/thailand-provinces.topojson');
https.get(url, function (res) {
    res.pipe(file);
    res.on('end', () => console.log('File successfully written!'));
}).on('error', (e) => {
    console.error(e);
});
